lib/linguist/heuristics.yml

```diff
@@ -619,6 +641,11 @@ named_patterns:
   - '^[ \t]*(class|(using[ \t]+)?namespace)\s+\w+'
   - '^[ \t]*(private|public|protected):$'
   - 'std::\w+'
+  euphoria:
+  - '^\s*(namespace)\s+'
+  - '^\s*(?:(public)\s+)?(include)\s+'
+  - '^\s*(?:(public|export|global)\s+)?(constant|enum|atom|integer|sequence|object)\s+'
+  - '^\s*(?:(public|export|global)\s+)?(function|procedure|type)\s+'
```

Comment:
Both of these expressions are very similar, so I recommend merging them into one so we avoid repeating the first half of the regex:

~~~suggestion
  - '^\s*(?:(?:public|export|global)\s+)?(?:atom|constant|enum|function|integer|object|procedure|sequence|type)\s'
~~~